CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+# Copyright 2022
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+cmake_minimum_required(VERSION 3.1)
+project(secp256k1 VERSION 0.1.0 LANGUAGES C)
+
+set(CMAKE_C_STANDARD 90)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS OFF)
+
+option(ENABLE_DEV_MODE "enable all binaries and modules by default but individual options can still be overridden explicitly" OFF)
+
+option(BUILD_BENCHMARK "compile benchmark" ON)
+option(BUILD_TESTS "compile tests" ON)
+option(BUILD_EXHAUSTIVE_TESTS "compile exhaustive tests" ON)
+option(BUILD_EXAMPLES "compile examples" ${ENABLE_DEV_MODE})
+
+option(ENABLE_MODULE_ECDH "enable ECDH module" ${ENABLE_DEV_MODE})
+option(ENABLE_MODULE_RECOVERY "enable ECDSA pubkey recovery module" ${ENABLE_DEV_MODE})
+option(ENABLE_MODULE_EXTRAKEYS "enable extrakeys module" ${ENABLE_DEV_MODE})
+option(ENABLE_MODULE_SCHNORRSIG "enable schnorrsig module" ${ENABLE_DEV_MODE})
+
+option(ALLOW_EXPERIMENTAL "allow experimental configure options" ${ENABLE_DEV_MODE})
+option(USE_EXTERNAL_DEFAULT_CALLBACKS "enable external default callback functions" OFF)
+option(COVERAGE "enable compiler flags to support kcov coverage analysis" ${ENABLE_DEV_MODE})
+
+set(ECMULT_WINDOW_SIZE "auto" CACHE STRING "window size for ecmult precomputation for verification, specified as integer in range [2..24]. \"auto\" is a reasonable setting for desktop machines (currently 15). [default=auto]")
+if(ECMULT_WINDOW_SIZE STREQUAL auto)
+  set(ECMULT_WINDOW_SIZE 15)
+endif()
+if(NOT ECMULT_WINDOW_SIZE MATCHES ^[1-9][0-9]*$ OR ECMULT_WINDOW_SIZE LESS 2 OR ECMULT_WINDOW_SIZE GREATER 24)
```

Comment:
Your suggestion checks for value validity, rather invalidity :)

If you mean to drop regexp, for some reasons (I did not investigate them) it will accept -DECMULT_WINDOW_SIZE=8a.

Reply:
Hm, I see. It probably uses something like the C functions `atoi()` or `atol()` under the hood; I think they simply ignoring trailing garbage. We could also convert it to a number via `math()` but yeah, this is not elegant either.



> Your suggestion checks for value validity, rather invalidity :)

ok yes but this could be fixed :)

Reply:
Maybe it would make sense to extract `MATCHES ^[1-9][0-9]*$` into a separate macro/function IS_NUM or similar and then use it for all numeric options. 

Just a suggestion, I don't have strong opinions, as long as the code does what it's supposed to do.

Reply:
I'd leave it as is for now. Code still concise and readable, imo :)